### PR TITLE
ARROW-12836: [C++] Add support for newer IBM i

### PR DESCRIPTION
**Pull request by mennosiek**
Newer versions of IBM i have CMAKE_SYSTEM_PROCESSOR set to powerpc or powerpc64. Hence in the current release ARROW_CPU_FLAG  defaults to "x86" which makes the install fail.

### CODE REVIEW

File: cpp/cmake_modules/SetupCxxFlags.cmake [hunk 1]
```diff
@@ -28,7 +28,7 @@ if(NOT DEFINED ARROW_CPU_FLAG)
     set(ARROW_CPU_FLAG "armv8")
   elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "armv7")
     set(ARROW_CPU_FLAG "armv7")
-  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "powerpc|powerpc64|ppc")
```
Comment (kou) [inline cpp/cmake_modules/SetupCxxFlags.cmake:31]: It seems that `poerpc64` is redundant.

Does the following pattern work?

```suggestion
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "powerpc|ppc")
```

Comment (mennosiek) [reply cpp/cmake_modules/SetupCxxFlags.cmake:31]: Yes, that will work.

Comment (kou) [reply cpp/cmake_modules/SetupCxxFlags.cmake:31]: OK. I'll apply it.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**